### PR TITLE
pmacct: 1.7.5 -> 1.7.6

### PR DESCRIPTION
**Pull request by 0x4A6F**
###### Motivation for this change
Upgrade to [1.7.6](https://github.com/pmacct/pmacct/releases/tag/v1.7.6) incorporating these [changes](https://github.com/pmacct/pmacct/compare/v1.7.5...v1.7.6).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/pmacct/default.nix [hunk 1]
```diff
@@ -1,55 +1,80 @@
...
+  ] ++ lib.optionals withNflog [
+    "--enable-nflog"
+  ] ++ lib.optionals withSQLite [
+    "--enable-sqlite3"
+  ] ++ lib.optionals withPgSQL [
+    "--enable-pgsql"
+  ] ++ lib.optionals withMysql [
+    "--enable-mysql"
+  ] ++ lib.optionals gnutlsSupport [
+    "--enable-gnutls"
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/pmacct/default.nix:76]: I think the old syntax was easier to overlook and shorter.

File: pkgs/tools/networking/pmacct/default.nix [hunk 2]
```diff
@@ -1,55 +1,80 @@
...
-, withMysql ? true, libmysqlclient ? null }:
+, withMysql ? true, libmysqlclient ? null
+, gnutlsSupport ? false, gnutls ? null
+}:
 
 assert withJansson -> jansson != null;
 assert withNflog -> libnetfilter_log != null;
 assert withSQLite -> sqlite != null;
 assert withPgSQL -> postgresql != null;
 assert withMysql -> libmysqlclient != null;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/pmacct/default.nix:25]: This is deprecated and should be removed.

File: pkgs/tools/networking/pmacct/default.nix [hunk 3]
```diff
@@ -1,55 +1,80 @@
...
 
 # Optional Dependencies
 , zlib ? null
 , withJansson ? true, jansson ? null
 , withNflog ? true, libnetfilter_log ? null
 , withSQLite ? true, sqlite ? null
 , withPgSQL ? true, postgresql ? null
-, withMysql ? true, libmysqlclient ? null }:
+, withMysql ? true, libmysqlclient ? null
+, gnutlsSupport ? false, gnutls ? null
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/pmacct/default.nix:17]: Those inputs can never be null unless overriden to that.

**Comment by 0x4A6F** [comment]
Thanks, updated.

File: pkgs/tools/networking/pmacct/default.nix [hunk 4]
```diff
@@ -1,55 +1,66 @@
...
+  ] ++ lib.optionals withPgSQL [
+    postgresql
+  ] ++ lib.optionals withMysql [
+    libmysqlclient
+    zlib
+  ] ++ lib.optionals gnutlsSupport [
+    gnutls
+  ];
 
   MYSQL_CONFIG =
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/pmacct/default.nix:53]:
```suggestion
```

File: pkgs/tools/networking/pmacct/default.nix [hunk 5]
```diff
@@ -1,55 +1,66 @@
...
-    optionalString withMysql "${getDev libmysqlclient}/bin/mysql_config";
+    lib.optionalString withMysql "${lib.getDev libmysqlclient}/bin/mysql_config";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/pmacct/default.nix:54]:
```suggestion
    MYSQL_CONFIG = lib.optionalString withMysql "${lib.getDev libmysqlclient}/bin/mysql_config";
```

File: pkgs/tools/networking/pmacct/default.nix [hunk 6]
```diff
@@ -1,55 +1,66 @@
...
+  ] ++ lib.optionals withSQLite [
+    sqlite
+  ] ++ lib.optionals withPgSQL [
+    postgresql
+  ] ++ lib.optionals withMysql [
+    libmysqlclient
+    zlib
+  ] ++ lib.optionals gnutlsSupport [
+    gnutls
+  ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/pmacct/default.nix:51]: I think the way it was before with lib.optional was easier to read and it is actual less code.

File: pkgs/tools/networking/pmacct/default.nix [hunk 7]
```diff
@@ -1,55 +1,66 @@
...
+  version = "1.7.6";
   pname = "pmacct";
 
   src = fetchFromGitHub {
     owner = "pmacct";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "17p5isrq5w58hvmzhc6akbd37ins3c95g0rvhhdm0v33khzxmran";
+    repo = "pmacct";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/pmacct/default.nix:26]:
```suggestion
    rev = version;
```

**Pull request merged**